include/grpcpp/ext/otel_plugin.h

```diff
@@ -26,24 +26,31 @@
 
 #include "absl/functional/any_invocable.h"
 #include "absl/status/status.h"
+#include "absl/status/statusor.h"
 #include "absl/strings/string_view.h"
 #include "opentelemetry/metrics/meter_provider.h"
 
 #include <grpc/support/metrics.h>
 #include <grpc/support/port_platform.h>
+#include <grpcpp/support/channel_arguments.h>
 
 namespace grpc {
 
 namespace internal {
 class OpenTelemetryPluginBuilderImpl;
-class OpenTelemetryPlugin;
 }  // namespace internal
 
 class OpenTelemetryPluginOption {
  public:
   virtual ~OpenTelemetryPluginOption() = default;
 };
 
+class OpenTelemetryPlugin {
+ public:
+  // Adds this OpenTelemetryPlugin to the channel args \a args.
+  virtual void AddToChannelArguments(grpc::ChannelArguments* args) = 0;
```

Comment:
probably should not be virtual?

Reply:
I was thinking - 
`grpc::OpenTelemetryPlugin` plugin be a non-abstract class with a private member `std::shared_ptr<grpc::internal::OpenTelemetry> plugin_`.

Reply:
> Yash, why is that better than the virtual method?

Yeah, we talked about this offline later, and I'm fine with the current approach as well. 
+1 to renaming `internal::OpenTelemetryPlugin` if sticking with the current approach.